src/Components/Admin/AssetModelViewer/assetModelViewer.js

```diff
@@ -57,15 +57,29 @@ export default function AssetModelViewer({
     const handleError = useErrorHandler();
     const [asset, setAsset] = useState({});
     const [name, setName] = useState("");
-    const [objectType, setObjectType] = useState("");
+    const [objectType, setObjectType] = useState(ObjectTypes.NONE);
     const [buttonText, setButtonText] = useState("Save");
     const [assetLink, setAssetLink] = useState("");
 
     useEffect(() => {
         const getObjectAsset = async () => {
             const data = await getAsset(assetId, handleError);
             setName(data.name);
-            setObjectType(data.obj_type);
+
+            switch (data.obj_type) {
+                case ObjectTypes.OBJECT:
+                    setObjectType(ObjectTypes.OBJECT);
+                    break;
+                case ObjectTypes.BACKGROUND:
+                    setObjectType(ObjectTypes.BACKGROUND);
+                    break;
+                default:
+                    setObjectType(ObjectTypes.NONE);
+                    throw new Error(
+                        "Object type must be either object or background"
```

Comment:
How do we handle this error? Technically we should never reach this, correct?

Reply:
Yea, we can only ever receive the values that are set in the selector drop down (object and background). Currently the error is not handled, but if need be I can add this in the future.